app/server.js

```diff
@@ -28,4 +38,214 @@ app.listen(PORT, () => {
 	console.log(`Server running on port ${PORT}`);
 });
 
+// Gets a list of brands
+app.get(BASE_URL + "/brands", (req, res) => {
+  try {
+    // Check if brands were retrieved successfully
+    if (!brands) return res.status(500).send(errorMessages[500]);
+    res.send(brands);
+  } catch(err){
+    next(err);
+  };
+});
+
+// Get a list of products with brand of provided id
+app.get(BASE_URL + "/brands/:id/products", (req, res) => {
+  try {
+    // Check if id parameter is present
+    const { id } = req.params;
+    if (!id) return res.status(400).send(errorMessages[400]);
+
+    // Create a list of brandIds and check if the id is included in it
+    const brandIds = brands.map((brand) => brand.id);
+    if (!brandIds.includes(id)) return res.status(404).send(errorMessages[404]);
+
+    // Filter products according to brandId
+    const filteredProduct = products.filter((product) => product.categoryId === id);
+
+    res.send(filteredProduct);
+  } catch(err) {
+    next(err);
+  };
+});
+
+// Get a list of all products
+app.get(BASE_URL + "/products", (req, res) => {
+  try{
+    // Check if products were retrieved successfully
+    if(!products) return res.status(500).send(errorMessages[500]);
+
+    res.send(products);
+  } catch(err) {
+    next(err);
+  };
+});
+
+// Login
+app.post(BASE_URL + "/login", (req,res) => {
+  try {
+
+    // log user in
+    const {username, password} = req.body;
+    const login = loginUser(username, password, users);
+    if (!login.loginSuccess) return res.status(login.err).send(login.message);
+
+    const user = login.user;
+
+    // Check if user already has access token
+    let userAccessToken = getAccessTokenFromUsername(user, ACCESS_TOKENS);
+
+    // If user did not already have accessToken make one for them and add to database
+    if (!userAccessToken) {
+      userAccessToken = AccessToken(username);
+      ACCESS_TOKENS.push(userAccessToken);
+    };
+
+    // Check if token is expired and if so update it
+    const tokenStillValid = isStillValid(userAccessToken);
+    if(!tokenStillValid) updateTimestamp(token);
+    
+    res.send(userAccessToken);
+  } catch (err) {
+    next(err);
+  }
+});
+
+// Get user cart
+app.get(BASE_URL + "/me/cart", (req, res) => {
+  try {
+    const token = req.header("token");
+
+    // Validate token
+    const tokenValidation = validateToken(token, ACCESS_TOKENS);
+    if (!tokenValidation.isValid) return res.status(tokenValidation.err).send(tokenValidation.message);
```

Comment:
DRY, you are repeating the same over and over in your authenticated apis. 
For an idea, check step 5 in here https://medium.com/@prashantramnyc/authenticate-rest-apis-in-node-js-using-jwt-json-web-tokens-f0e97669aad3